components/FormFlashcard/FormFlashcard.js

```diff
@@ -11,12 +13,47 @@ export default function FormFlashcard({
   currentFlashcard,
   onSubmitFlashcard,
   isFormClosing,
+  changeCollections,
 }) {
+  const [showNewCollectionFields, setShowNewCollectionFields] = useState(false);
+
+  function handleCollectionChange(event) {
+    if (event.target.value === "newCollection") {
+      setShowNewCollectionFields(true);
+    } else {
+      setShowNewCollectionFields(false);
+    }
+  }
+
+  function addNewCollection(formData) {
+    const newCollectionName = formData.get("collectionName");
+    const newCollectionColor = formData.get("collectionColor");
+    const newCollection = {
+      id: uid(),
+      title: newCollectionName,
+      color: newCollectionColor,
+    };
+    changeCollections(newCollection);
+    return newCollection.id;
+  }
+
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
-    const newFlashcard = Object.fromEntries(formData);
+    let newFlashcard;
+    if (showNewCollectionFields) {
+      const newId = addNewCollection(formData);
+      newFlashcard = {
+        collectionId: newId,
+        question: formData.get("question"),
+        answer: formData.get("answer"),
+      };
+    } else {
+      newFlashcard = Object.fromEntries(formData);
+    }
+
```

Comment:
I suggest to improve the clarity and structure of the `handleSubmit` function by first extracting all the form data and then processing it based on whether new collection fields are visible. This avoids calling `formData.get()` multiple times.
This way your code is more readable and ensures consistency when accessing form fields. Additionally, the logic is more streamlined and easier to follow.
I suggest to destructure `collectionName`, `collectionColor`, `question`,` answer`, and `collectionId` from `formData` to make the code cleaner and avoid multiple lookups.

```js
unction handleSubmit(event) {
  event.preventDefault();

  // Extract form data into a single object
  const formData = Object.fromEntries(new FormData(event.target));
  const { collectionName, collectionColor, question, answer, collectionId } = formData;

  // Initialize the new flashcard
  let newFlashcard;

  // If user adds a new collection
  if (showNewCollectionFields) {
    const newCollection = {
      id: uid(),
      title: collectionName,
      color: collectionColor,
    };
    changeCollections(newCollection);

    newFlashcard = {
      collectionId: newCollection.id,
      question,
      answer,
    };
  } else {
    // If no new collection is added, use existing form data
    newFlashcard = { ...formData };
  }

  onSubmitFlashcard(newFlashcard);
  setShowNewCollectionFields(false);
  event.target.reset();
}
```

Reply:
Good idea, thanks! I had a look at these two functions and thought that there must be an easier way to do it, but I didn't know how because, firstly, I need the newly generated collections.id in the newly created flashcard and, secondly, I don't need all the data from formData in the flashcard when a new collection is created. Destructuring formData is a great solution (and actually simple), which I hadn't thought of. Thanks! :pray: